houndigrade/cli.py

```diff
@@ -0,0 +1,207 @@
+"""Command line script for inspecting attached volumes."""
+import glob
+import json
+import os
+import subprocess
+from gettext import gettext as _
+
+import click
+from kombu import Connection, Exchange, Queue
+
+INSPECT_PATH = '/mnt/inspect'
+
+
+@click.command()
+@click.option('--cloud',
+              '-c',
+              default='aws',
+              help='Cloud in which we are performing the inspection.',
+              type=click.Choice(['aws', 'gcp', 'azure']))
+@click.option('--target',
+              '-t',
+              multiple=True,
+              type=(str, click.Path(exists=True)),
+              required=True,
+              help=_('Inspection target, the cloud specific image identifier'
+                     ' and path to the attached drive on the machine. e.g.'
+                     ' -t ami-12312839312 /dev/sda'))
+@click.option('--debug',
+              is_flag=True,
+              help=_('Print debug output.'))
+def main(cloud, target, debug):
+    r"""
+    Mounts provided volumes and inspects them.
+
+    The script takes provided target information and checks if there are any
+    partitions on that device, then it loops through, mounting every partition
+    and inspecting it. As it loops through it builds a dictionary of results
+    that gets placed on a queue once the processing is done.
+
+    \b
+    Args:
+        cloud (str): Cloud in which we are performing the inspection.
+        target (str): Inspection target, the cloud specific image identifier
+                      and path to the attached drive on the machine.
+        debug (boolean): Boolean flag that turns on debug output.
+
+    """
+    click.echo(_('Provided cloud: {}'.format(cloud)))
+    click.echo(_('Provided drive(s) to inspect: {}'.format(target)))
+
+    results = {
+        'cloud': cloud,
+        'inspection_targets': target,
+        'facts': {},
+    }
+
+    for image_id, drive in target:
+        click.echo('Checking drive {}'.format(drive))
```

Comment:
What's the reason for using `click.echo` instead of the standard Python logger? Does `echo` give us the option to include formatting, timestamp, etc. in its output? Do you think there would be any value in having a richer output format that could be given by the Python logger, or will we only ever want bare-bones stdout from this script?

(FWIW, I'm not terribly familiar with the click library, but I did look enough into the docs to see that `echo` appears to be a thin wrapper around `print` that can handle things like ANSI color codes.)

Reply:
imo even the stdout is overkill and I've considered removing it entirely, or putting all of it behind the `--debug` flag since normally where this will be running will never really be seen by anyone. There was no specific reason for using `click.echo`, but it does offer few things over standard print (like being able to easily output to stderr), and ability to use its other functions later on if deemed necessary (things like progress bars, pagers, colors, formatting path names, etc.).

As for the standard python logger ¯\\\_(ツ)\_/¯ I don't see a lot of value since we're running in a container without having to spend additional time to set up another tool that would consume those logs so they wouldn't disappear when the container exits.